src/main/java/yuki/Yuki.java

```diff
@@ -0,0 +1,229 @@
+            throw new YukiExceptions.InvalidIndexException("Invalid index for marking: " + indexTask);
+        }
+        if (cmd.equals(Constants.MARK_COMMAND)){
+            tasks.get(indexTask - 1).markAsDone();
+        } else if (cmd.equals(Constants.UNMARK_COMMAND)) {
+            tasks.get(indexTask - 1).markAsUndone();
+        } else {
+            System.out.println("invalid command in mark method");
+        }
+    }
+
+    public static void deleteTask(String line) throws YukiExceptions.InvalidIndexException {
+        int indexTask = Integer.parseInt(line.split(" ")[1]);
+        if (indexTask < 0 || indexTask > tasks.size()) {
+            throw new YukiExceptions.InvalidIndexException("Invalid index for marking: " + indexTask);
+        }
+        tasks.remove(indexTask - 1);
+    }
+
+    public static void addTodo(String input) throws YukiExceptions.InvalidDescriptionException {
+        data = InputParser.parseInput(input.substring(Constants.LENGTH_TODO_COMMAND));
+        if (data[0].isEmpty()) {
+            throw new YukiExceptions.InvalidDescriptionException("Invalid description");
+        }
+        description = data[0];
+        Task t = new Todo(description);
+        tasks.add(t);
+        System.out.println(t);
+        reportNumberOfTasks();
+    }
+
+    public static void addDeadline(String input) throws YukiExceptions.InvalidDescriptionException {
+        data = InputParser.parseInput(input.substring(Constants.LENGTH_DEADLINE_COMMAND));
+        if (data.length < 2) {
+            throw new YukiExceptions.InvalidDescriptionException("Invalid description");
+        }
+        description = data[0] + " (by:" + data[1] + ")";
+        Task t = new Deadline(description);
+        tasks.add(t);
+        System.out.println(t);
+        reportNumberOfTasks();
+    }
+
+    public static void addEvent(String input) throws YukiExceptions.InvalidDescriptionException {
+        data = InputParser.parseInput(input.substring(Constants.LENGTH_EVENT_COMMAND));
+        if (data.length < 3) {
+            throw new YukiExceptions.InvalidDescriptionException("Invalid description");
+        }
+        description = data[0] + " (from: " + data[1] + " to: " + data[2] + ")";
+        Task t = new Event(description);
+        tasks.add(t);
+        System.out.println(t);
+        reportNumberOfTasks();
+    }
+
+    public static void reportNumberOfTasks() {
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void main(String[] args) {
+        try {
+            readFile();
+        } catch (FileNotFoundException e) {
+            System.out.println("Txt file of tasks not found, will create one");
+        }
+
+        Utils.printWelcomeMessage();
+
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+
+        String command;
+
+        while (!line.equals(Constants.EXIT_COMMAND)) {
```

Comment:
This while loop is very long, consider abstracting it further to break it down.

Reply:
In the TaskList.java > executeCommand method, I would like to reduce the number of try, catch statents - do you have advice on how I can do so?

![Screenshot 2024-02-24 at 8 50 35 PM](https://github.com/nus-cs2113-AY2324S2/ip/assets/72074553/186213aa-3398-439f-a17e-0887c886cbb1)